app/backend/src/tests/test_bg_jobs.py

```diff
@@ -938,6 +943,150 @@ def test_send_reference_reminders(db):
+            last_sent_request_reminder_time=now() - HOST_REQUEST_REMINDER_INTERVAL + timedelta(hours=1),
+        )
+
+        # case 4: start date is today => do not notify
+        hr4 = create_host_request_by_date(
+            session=session,
+            surfer_user_id=user7.id,
+            host_user_id=user8.id,
+            from_date=today(),
+            to_date=today() + timedelta(days=2),
+            status=HostRequestStatus.pending,
+            host_sent_request_reminders=0,
+            last_sent_request_reminder_time=now() - HOST_REQUEST_REMINDER_INTERVAL,
+        )
+
+        # case 5: from_date in the past => do not notify
+        hr5 = create_host_request_by_date(
+            session=session,
+            surfer_user_id=user9.id,
+            host_user_id=user10.id,
+            from_date=today() - timedelta(days=1),
+            to_date=today() + timedelta(days=1),
+            status=HostRequestStatus.pending,
+            host_sent_request_reminders=0,
+            last_sent_request_reminder_time=now() - HOST_REQUEST_REMINDER_INTERVAL,
+        )
+
+        # case 6: non-pending status => do not notify
+        hr6 = create_host_request_by_date(
+            session=session,
+            surfer_user_id=user11.id,
+            host_user_id=user12.id,
+            from_date=today() + timedelta(days=3),
+            to_date=today() + timedelta(days=4),
+            status=HostRequestStatus.accepted,
+            host_sent_request_reminders=0,
+            last_sent_request_reminder_time=now() - HOST_REQUEST_REMINDER_INTERVAL,
+        )
+
+        # case 7: host already sent a message => do not notify
+        hr7 = create_host_request_by_date(
+            session=session,
+            surfer_user_id=user13.id,
+            host_user_id=user14.id,
+            from_date=today() + HOST_REQUEST_REMINDER_INTERVAL + timedelta(days=1),
+            to_date=today() + HOST_REQUEST_REMINDER_INTERVAL + timedelta(days=2),
+            status=HostRequestStatus.pending,
+            host_sent_request_reminders=0,
+            last_sent_request_reminder_time=now() - HOST_REQUEST_REMINDER_INTERVAL,
+        )
+
+        session.add(
+            Message(
+                time=now(),
+                conversation_id=hr7,
+                author_id=user14.id,
+                text="Looking forward to hosting you!",
+                message_type=MessageType.text,
+            )
+        )
+
+    send_host_request_reminders(empty_pb2.Empty())
+
+    while process_job():
+        pass
+
+    with session_scope() as session:
+        emails = [
+            (email.recipient, email.subject, email.plain, email.html)
+            for email in session.execute(select(Email).order_by(Email.recipient.asc())).scalars().all()
+        ]
+
+    expected_emails = [
+        (
+            "user2@couchers.org.invalid",
+            "[TEST] You have a pending host request from User 1!",
+            ("Please respond to the request!",),
```

Comment:
Please include the surfer user's info in a test string to verify that last comment